api/src/pcapi/core/offers/validation.py

```diff
@@ -571,6 +572,31 @@ def check_offer_extra_data(
         raise errors
 
 
+def check_product_from_ean(
+    ean_code: str | None,
+    product_id: int | None,
+    product: models.Product | None,
+    subcategory_id: str | None,
+    venue_type_code: VenueTypeCode,
+) -> None:
+    if ean_code is not None or product_id is not None:
+        if product:
+            return None
```

Comment:
ça devrait aller sans le None maintenant que ton typage est juste None
```suggestion
            return
```

Reply:
Yes en effet